src/components/auth/signup.js

```diff
@@ -0,0 +1,122 @@
+import React, { Component } from 'react';
+import { Field, reduxForm } from 'redux-form';
+import { Button, Form, FormGroup, FormFeedback, Label, Input } from 'reactstrap';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+
+import CarouselComponent from '../universal/carousel';
+import { signUpAction } from '../../actions';
+
+class Signup extends Component {
+
+    // method for rendering form field on the component
+    renderField(field) {
+        //de-structuring from the field object
+        const { meta: { touched, error } } = field;
+        const isValid = touched && error ? false : '';
+        return (
+            <FormGroup>
+                <Label htmlFor={ field.htmlfor }>{ field.label }</Label>
+                <Input 
+                    type={ field.type }
+                    id={ field.id }
+                    { ...field.input }
+                    valid = { isValid }
+                />
+                <FormFeedback>{ touched ? error : '' }</FormFeedback>
+            </FormGroup>
+        );
+    }
+
+    // method callback on form submission
+    onSubmitForm = (values) => {
+        this.props.signUpAction(values, () => {
+            this.props.history.push('/signin');
+        });
+
+    }
+
+    render() {
+        const { handleSubmit } = this.props;
+        return (
+            <div>
+                <CarouselComponent />
+                <div className="container">
+                    <div class="row justify-content-md-center">
+                        <div className="col-md-6">
+                            <h3>CREATE ACCOUNT</h3>
+                            <hr />
+                            <Form onSubmit={ handleSubmit(this.onSubmitForm.bind(this)) }>
```

Comment:
again, no need to re-bind `onSubmitForm` function since you used an arrow function to define it above